### PR TITLE
New Adapter mabidder

### PR DESCRIPTION
**Pull request by ecdrsvc**
We have an existing Prebid.js adapter and would like to add this new Prebid Server adapter.

### CODE REVIEW

**Comment by ecdrsvc** [comment]
Documentation PR: https://github.com/prebid/prebid.github.io/pull/4848

**Comment by onkarvhanumante** [comment]
@ecdrsvc  feature branch [ecdrsvc:master](https://github.com/ecdrsvc/prebid-server/tree/master) is  [22 commits ahead of](https://github.com/ecdrsvc/prebid-server/compare/prebid%3Aprebid-server%3Amaster...master), [204 commits behind](https://github.com/ecdrsvc/prebid-server/compare/master...prebid%3Aprebid-server%3Amaster)

should sync feature master with latest [prebid:master](https://github.com/prebid/prebid-server/tree/master) updates.

**Comment by ecdrsvc** [comment]
@onkarvhanumante, it's synced.

File: adapters/mabidder/mabidder_test.go
Comment (onkarvhanumante) [inline adapters/mabidder/mabidder_test.go:14]: should use `https://prebid.ecdrsvc.com/pbs` url as mentioned in bidder-info param

https://github.com/prebid/prebid-server/blob/183345e49f0ff04bf9e6044dfa93b8d9ccffe9f3/static/bidder-info/mabidder.yaml#L1

File: adapters/mabidder/mabidder.go
Comment (onkarvhanumante) [inline adapters/mabidder/mabidder.go:84]: just a suggestion, could make use of utils mentioned below

https://github.com/prebid/prebid-server/blob/06151091ee480fefa42a25020738f41ec1902333/adapters/response.go#L9-L26

File: adapters/mabidder/mabidder.go
Comment (onkarvhanumante) [inline adapters/mabidder/mabidder.go:38]: scope of these struct is within `mabidder`  package. so could `ma` prefix from above structs
```
- type maServerResponse struct {
+ type serverResponse struct {

- type maBidResponse struct {
+ type bidResponse struct {

-  type maMeta struct {
+  type meta struct {
```

File: adapters/mabidder/mabidder.go
Comment (onkarvhanumante) [inline adapters/mabidder/mabidder.go:109]: `bidResponse` is initialized  by `NewBidderResponseWithBidsCapacity()` on line 90

https://github.com/prebid/prebid-server/blob/8ffa1c66082f1d289aa17206e9b54668afb14741/adapters/bidder.go#L72-L78

Based on code setup, `bidResponse.Currency` will always be `USD`. Therefore, `if` check on line 107 will always be false and Currency from `maBidResp` won't be used. Perhaps you would like to make following change.

```
	if maBidResp.Currency != "" {
		bidResponse.Currency = maBidResp.Currency
	}
```

File: static/bidder-info/mabidder.yaml
Comment (onkarvhanumante) [inline static/bidder-info/mabidder.yaml:1]: endpoint is reachable

File: static/bidder-info/mabidder.yaml
Comment (onkarvhanumante) [inline static/bidder-info/mabidder.yaml:3]: @ecdrsvc , PBS team attempted to verify above maintainer email. However PBS team was not able to reach you out on above specified email address.  

Note that an working maintainer email is needed to proceed with this PR

**Review by gargcreation1992 — COMMENTED** [review]
g2g after addressing Onkar's comment.

**Comment by ecdrsvc** [comment]
@onkarvhanumante the code has been updated with all your suggestions.
Regarding the email, there seems to be an issue with it. I'm waiting to hear back from our ops team and will update here when it's resolved.

**Review by gargcreation1992 — APPROVED** [review]
g2g after email verification.

**Comment by onkarvhanumante** [comment]
> @onkarvhanumante the code has been updated with all your suggestions. Regarding the email, there seems to be an issue with it. I'm waiting to hear back from our ops team and will update here when it's resolved.

adding block label on PR until we get update on working email

**Comment by ecdrsvc** [comment]
> > @onkarvhanumante the code has been updated with all your suggestions. Regarding the email, there seems to be an issue with it. I'm waiting to hear back from our ops team and will update here when it's resolved.
> 
> adding block label on PR until we get update on working email

The email issue has been fixed. Please send a verification message and we will reply. Thanks.

**Comment by onkarvhanumante** [comment]
> > > @onkarvhanumante the code has been updated with all your suggestions. Regarding the email, there seems to be an issue with it. I'm waiting to hear back from our ops team and will update here when it's resolved.
> > 
> > 
> > adding block label on PR until we get update on working email
> 
> The email issue has been fixed. Please send a verification message and we will reply. Thanks.

Verification email is being sent to `lmprebidadapter@loblaw.ca`. Waiting for reply

File: static/bidder-info/mabidder.yaml
Comment (gargcreation1992) [reply static/bidder-info/mabidder.yaml:3]: sent an email again. Please reply with "received"

**Review by gargcreation1992 — APPROVED** [review]
g2g after email verification.

**Comment by ecdrsvc** [comment]
Just confirming here that we got the verification email and replied with 'received'. Let me know if any issues.

**Comment by onkarvhanumante** [comment]
> Just confirming here that we got the verification email and replied with 'received'. Let me know if any issues.

received reply from mabidder

**Review by onkarvhanumante — APPROVED** [review]

**Pull request merged**
